sdk/database/dbplugin/v5/grpc_server.go

```diff
@@ -173,7 +271,15 @@ func (g gRPCServer) Type(ctx context.Context, _ *proto.Empty) (*proto.TypeRespon
 }
 
 func (g gRPCServer) Close(ctx context.Context, _ *proto.Empty) (*proto.Empty, error) {
-	err := g.impl.Close()
+	g.Lock()
+	defer g.Unlock()
+
+	impl, err := g.getDatabaseInternal(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = impl.Close()
 	if err != nil {
 		return &proto.Empty{}, status.Errorf(codes.Internal, "unable to close database plugin: %s", err)
 	}
```

Comment:
After closing, we should also deleting the instance from the `instances` map.

Reply:
addressed in: https://github.com/hashicorp/vault/pull/13734/commits/e551043fb368569e1a7c643270854cacb3f41d48